src/external.py

```diff
@@ -0,0 +1,188 @@
+
+    Args:
+        interface (hoomd.variant.variant): position of the interface
+        geometry (str): Drying geometry ("film" or "droplet")
+
+    Args:
+        k (float): Harmonic coefficient
+
+        offset (float): per-particle-type amount to shift in the potential
+
+    """
+
+    def __init__(self, interface=None, geometry=None):
+        super().__init__()
+
+        if geometry not in ("film", "droplet"):
+            raise ValueError(f"Unrecognized geometry: {geometry}")
+
+        param_dict = ParameterDict(interface=hoomd.variant.Variant, geometry=str)
+        param_dict["interface"] = interface
+        param_dict["geometry"] = geometry
+        param_dict.update(
+            dict(
+                interface=interface,
+                geometry=geometry,
+            )
+        )
+        self._param_dict.update(param_dict)
+
+        params = TypeParameter(
+            "params",
+            "particle_types",
+            TypeParameterDict(k=float, offset=float, len_keys=1),
+        )
+        self._add_typeparam(params)
+
+    def _attach_hook(self):
+        sim = self._simulation
+
+        cpp_class = getattr(_azplugins, self._make_cpp_class_name())
+
+        if self.geometry not in ("film", "droplet"):
+            raise ValueError(f"Unrecognized geometry: {self.geometry}")
+
+        self._cpp_obj = cpp_class(sim.state._cpp_sys_def, self.interface)
+
+        super()._attach_hook()
+
+    def _make_cpp_class_name(self):
+        cpp_class_name = self.__class__.__name__
+        if isinstance(self._simulation.device, hoomd.device.GPU):
+            cpp_class_name += "GPU"
+        return cpp_class_name
+
+
+class PlanarHarmonicBarrier(HarmonicBarrier):
+    r"""Applies a purely repulsive harmonic potential to particles
+        near an interface in planar geometry..
+
+    Args:
+        interface (`hoomd.variant`) : `z` position of the interface
+
+    `PlanarHarmonicBarrier` apply a purely harmonic potential in planar geometry
+    that pushes down on nonvolatile solutes. The potential is truncated at its minimum
+    and have following form:
+
+    .. math::
+
+        \begin{eqnarray*}
+        V(d) = & 0 & d < H \\
+               & \frac{\kappa}{2} (d-H)^2 & H < d
+        \end{eqnarray*}
+
+    Here, the interface is located at height *H*, and may change with time.
+    The effective interface position *H* may be modified per-particle-type
+    using a *offset* (*offset* is added to *H* to determine the effective *H*)
+    :math:`\kappa` is a spring constant setting the strength of the interface
+    (:math:`\kappa` is a proxy for the surface tension).
+
+    Example::
+
+        interf = hoomd.variant.Ramp(A=50.0, B=25.0, t_start=100, t_ramp=1e6)
+        evap = hoomd.azplugins.external.PlanarHarmonicBarrier(interface = interf)
+
+        evap.params['S'] = dict(k=50.0, offset=0.0)
+        evap.params['B'] = dict(k=50.0*2*2, offset=0.0)
```

Comment:
```suggestion
        evap = hoomd.azplugins.external.PlanarHarmonicBarrier(interface=interf)
        evap.params['S'] = dict(k=50.0, offset=0.0)
        evap.params['B'] = dict(k=50.0*2**2, offset=0.0)
```
Please also add a short description of what this example is implementing.